### ISSUE TITLE
TPCH distributed test errors

### ISSUE THREAD

**Issue by arzuschen** [zhouqingqing/qpmodel#188]
Currently, only a fraction of the tpch queries can be run
(q02, q04, q07, q08, q09, q13, q14, q15, q17, q18, q20, q21, q22 cannot be run)

Also, q10 has wrong output and q03 has error in cardinality estimation. (row = 0 while not 0 in non distributed plan)

**Comment by arzuschen** [comment]
non-cached subquery, which requires parameters from main query, cannot be executed in the distributed

**Comment by zhouqingqing** [comment]
1. broadcast optimization;
2. fix aggr by splitting global/local;
3. add more primitive tests;
4. document with memo print out;

**Comment by arzuschen** [comment]
q07, q08, q09, q22 are caused by gather not finding distributed context if placed under physicfromquery
q13 is caused by memo not getting any possible solution
q15 debug assertion not passed since selectstmt.distributed_ changed more than once

(q13 concerns optimizer and q15 should be a simple fix, I will work on those two first)

**Comment by 9DemonFox** [comment]
> q07, q08, q09, q22 are caused by gather not finding distributed context if placed under physicfromquery
fixed, see  PR#264.
